src/vector_v4.hpp

```diff
@@ -14,8 +14,7 @@ template <typename T> class VectorV4 {
         for (size_t i = 0; i < size_; i++) {
             temp[i].~T();
         }
```

Comment:
Your destroy is problematic. Same problem as Optional.

Test:
In a scope {
 - Create a vector.
 - Push 1 elements in it.
 - call destroy on it.
}

It will crash.